middleware/authentication.js

```diff
@@ -0,0 +1,25 @@
+const User = require('../models/user')
+const jwt = require('jsonwebtoken')
+const { UnauthenticatedError } = require('../errors/UnauthenticatedError')
+
+const auth = async (req, res) => {
+    const authHeader = req.headers.authorization
+    if (!authHeader || !authHeader.startsWith('Bearer')) {
+        throw new UnauthenticatedError('Authentication invalid')
+    }
+    const token = authHeader.split(' ')[1]
+
+    try {
+        const payload = jwt.verify(token, process.env.JWT_SECRET)
+        const expiry = payload.exp
+        if (expiry < Date.now()) {
+            throw new UnauthenticatedError('Authentication invalid')
+        }
+
+        req.user = { userId: payload.userId, name: payload.username }
```

Comment:
Does this line add the user onto the req so the services can access the userId that way? Thats cool if so